pkg/noderesourcetopology/filter_test.go

```diff
@@ -0,0 +1,461 @@
+
+func (m mockIndexer) Add(interface{}) error               { return nil }
+func (m mockIndexer) Update(obj interface{}) error        { return nil }
+func (m mockIndexer) Delete(obj interface{}) error        { return nil }
+func (m mockIndexer) List() []interface{}                 { return nil }
+func (m mockIndexer) ListKeys() []string                  { return nil }
+func (m mockIndexer) Replace([]interface{}, string) error { return nil }
+func (m mockIndexer) Resync() error                       { return nil }
+func (m mockIndexer) Get(obj interface{}) (item interface{}, exists bool, err error) {
+	return nil, false, nil
+}
+
+func (m mockIndexer) GetByKey(key string) (item interface{}, exists bool, err error) {
+	if v, ok := m.nodeTopologies[key]; ok {
+		return &v, ok, nil
+	}
+	return nil, false, fmt.Errorf("Node topology is not found: %v", key)
+}
+
+func makeResourceListFromZones(zones topologyv1alpha1.ZoneList) v1.ResourceList {
+	result := make(v1.ResourceList)
+	for _, zone := range zones {
+		for _, resInfo := range zone.Resources {
+			resQuantity, err := resource.ParseQuantity(resInfo.Allocatable.String())
+			if err != nil {
+				klog.Errorf("Failed to parse %s", resInfo.Allocatable.String())
+				continue
+			}
+			if quantity, ok := result[v1.ResourceName(resInfo.Name)]; ok {
+				resQuantity.Add(quantity)
+			}
+			result[v1.ResourceName(resInfo.Name)] = resQuantity
+		}
+	}
+	return result
+}
+
+func makePodByResourceListWithManyContainers(resources *v1.ResourceList, containerCount int) *v1.Pod {
+	containers := []v1.Container{}
+
+	for i := 0; i < containerCount; i++ {
+		containers = append(containers, v1.Container{
+			Resources: v1.ResourceRequirements{
+				Requests: *resources,
+				Limits:   *resources,
+			},
+		})
+	}
+	return &v1.Pod{Spec: v1.PodSpec{Containers: containers}}
+}
+
+func runTests(t *testing.T, topologyTests topologyMatchTests) {
+	nodeInfo := framework.NewNodeInfo()
+	for _, test := range topologyTests {
+		t.Run(test.name, func(t *testing.T) {
+			tm := TopologyMatch{
+				policyHandlers: PolicyHandlerMap{
+					topologyv1alpha1.SingleNUMANodePodLevel:       SingleNUMAPodLevelHandler,
+					topologyv1alpha1.SingleNUMANodeContainerLevel: SingleNUMAContainerLevelHandler,
+				},
+				namespaces: []string{metav1.NamespaceDefault},
+				lister:     listerv1alpha1.NewNodeResourceTopologyLister(mockIndexer{nodeTopologies: test.nodeTopologies}),
+			}
+			nodeInfo.SetNode(&test.node)
+			test.pod.Spec.Containers[0].Name = containerName
+			gotStatus := tm.Filter(context.Background(), framework.NewCycleState(), test.pod, nodeInfo)
+
+			if !reflect.DeepEqual(gotStatus, test.wantStatus) {
+				t.Errorf("status does not match: %v, want: %v\n", gotStatus, test.wantStatus)
+			}
+		})
+	}
+}
+
+func TestTopologyRequestsCorrectNodeResourceTopology(t *testing.T) {
+	nodes := nodeTopologyMap{}
+	nodes["default/node1"] = topologyv1alpha1.NodeResourceTopology{
```

Comment:
It seems only the cpu portions are different, so could you dedupe the code below?